### PR TITLE
Adds a doc on deferred processing.

### PR DESCRIPTION
**Pull request by jdoliner**

### CODE REVIEW

File: doc/cookbook/deferred_processing.md [hunk 1]
```diff
@@ -2,50 +2,58 @@
...
+a staging area to then process later.
 
 Commit a file to the staging branch:
 
 ```sh
 $ pachctl put-file data staging -f <file>
 
-# Note: Doing put-file to a branch that doesn't exist, in this case "stages", will automatically create that branch. 
+# Note: Doing put-file to a branch that doesn't exist, in this case
+"stages", will automatically create that branch.
```
Comment (JoeyZwicker) [inline doc/cookbook/deferred_processing.md]: Minor typo on my part: should be "staging"

**Comment by gabrielgrant** [comment]
Why is `set-branch` being deprecated? those examples seem like they'd be more readable with `set` (or `update`) rather than `create`

**Comment by jdoliner** [comment]
`set-branch`'s doesn't match with the rest of our naming scheme. We use `create` for all other objects, and `put/get` for requests that take/return streams of bytes. We have an eventual plan to move to a more kubernetes style CLI where you'd have a subcommand `create` which you can then use to create various objects. It'd be really weird at that point to have a subcommand `set` which only had branch under it. There's also the fact that `set-branch` can actually be used to create branches, in which case I think it's a slightly confusing name. When the branch already exists I agree that it's a more confusing name, but I think the consistency makes it worth it. We could maybe add `update-branch` like we have `update-pipeline` and `update-repo`, which are really just `create-pipeline` and `create-repo` under the hood.

**Comment by JoeyZwicker** [comment]
Just commenting here for future reference and then I'll make the adjustment. When adding new cookbooks or docs sections, it needs to be added to the table of contents (ToC) in the index file otherwise it wont show up in the docs sidebar

**Comment by gabrielgrant** [comment]
@jdoliner yes, to be clear -- i agree `update` would be the most consistent and readable

**Pull request merged**